pyatlan/generator/templates/entity.jinja2

```diff
@@ -925,6 +961,8 @@ class {{ entity_def.name }}({{super_classes[0]}} {%- if "Asset" in super_classes
                 name=name, anchor=anchor, glossary_qualified_name=glossary_qualified_name,  glossary_guid = glossary_guid, categories=categories
             )
         )
+    def trim_to_required(self) -> AtlasGlossaryTerm:
+        return self.create_for_modification(qualified_name=self.qualified_name, name=self.name, glossary_guid = self.anchor.guid)
```

Comment:
As above — extra check for `self.anchor` before trying to resolve `self.anchor.guid` (?)